Refactor/code cleanup

AnaqiAmir opened the pull request.

AnaqiAmir committed:
refactor: clean up EDA

1) cleanup EDA section
2) add  ucimlrepo in requirements.txt

AnaqiAmir committed:
feat: impute missing values

Applied label encoding, OHE, and scaler

AnaqiAmir committed:
feat: add categorical features EDA

AnaqiAmir committed:
style: consolidate model comparison

AnaqiAmir committed:
style: cleanup evaluation, tuning,  conclusion

AnaqiAmir committed:
doc: improve documentation

AnaqiAmir committed:
fix: edit LabelEncoder and KNNImputer processes

1) assign n_neighbors = 1
2) remove redundant label encoding assignments (remove 11 --> 5)
3) save new best model

Aditya-k-23 reviewed (approved):
LGTM with following changes
- Now that we use KNNImputation n_neighbours=1, need to change that in the comment above
- Need to change the values we get for XGB for all metrics in the conclusion block, current values are incorrect.
- Need to add missingno to requirements.txt 
- Make necessary changes to findings in the readme

AnaqiAmir committed:
doc: update docs

Update the following files with new model and metrics:
1) README.md
2) analysis.ipynb
3) Confusion-Matrix.png
4) Feature-Importance.png
5) requirements.txt
